client/service_create.go

```diff
@@ -62,9 +62,11 @@ func imageWithDigestString(image string, dgst digest.Digest) string {
 	if err == nil {
 		if _, isCanonical := ref.(reference.Canonical); !isCanonical {
 			namedRef, _ := ref.(reference.Named)
```

Comment:
The reason it was okay to assume (here) was that this part is only used if `cli.DistributionInspect` does not return an error. Still, I agree that it's good to make this function more robust.